inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+                             node->get_type_name(), ") ", "is not supported.");
+
+                pugi::xml_node port = input.append_child("port");
+                port.append_attribute("id").set_value(port_id++);
+                for (auto d : i.get_shape()) {
+                    pugi::xml_node dim = port.append_child("dim");
+                    dim.append_child(pugi::xml_node_type::node_pcdata)
+                        .set_value(std::to_string(d).c_str());
+                }
+            }
+        }
+        // <layers/output>
+        if ((node->get_output_size() > 0) &&
+            !dynamic_cast<ngraph::op::Result*>(node)) {
+            pugi::xml_node output = layer.append_child("output");
+            for (auto o : node->outputs()) {
+                NGRAPH_CHECK(o.get_partial_shape().is_static(),
+                             "Dynamic shape of ", node->get_name(), " (",
+                             node->get_type_name(), ") ", "is not supported.");
+
+                pugi::xml_node port = output.append_child("port");
+                port.append_attribute("id").set_value(port_id++);
+                port.append_attribute("precision")
+                    .set_value(get_output_precision_name(o).c_str());
+                for (auto d : o.get_shape()) {
+                    pugi::xml_node dim = port.append_child("dim");
+                    dim.append_child(pugi::xml_node_type::node_pcdata)
+                        .set_value(std::to_string(d).c_str());
+                }
+            }
+        }
+    }
+    // <edges>
+    const std::vector<Edge> edge_mapping = create_edge_mapping(layer_ids, f);
+    pugi::xml_node edges = netXml.append_child("edges");
+    for (auto e : edge_mapping) {
+        pugi::xml_node edge = edges.append_child("edge");
+        edge.append_attribute("from-layer").set_value(e.from_layer);
+        edge.append_attribute("from-port").set_value(e.from_port);
+        edge.append_attribute("to-layer").set_value(e.to_layer);
+        edge.append_attribute("to-port").set_value(e.to_port);
+    }
+}
+
+}  // namespace
+
+// ! [function_pass:serialize_cpp]
+// serialize.cpp
+bool pass::Serialize::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    // prepare data
+    pugi::xml_document xml_doc;
+    std::vector<uint8_t> constants;
+    switch (m_version) {
+    case Version::IR_V10:
+        ngfunction_2_irv10(xml_doc, constants, *f);
+        break;
+    default:
+        NGRAPH_UNREACHABLE("");
+        break;
+    }
+
+    // create xml file
+    std::ofstream xml_file(m_xmlPath, std::ios::out);
+    xml_doc.save(xml_file);
+
+    // create bin file
+    std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
+    bin_file.write(reinterpret_cast<const char*>(constants.data()),
+                   constants.size() * sizeof(constants[0]));
+
+    // Return false because we didn't change nGraph Function
+    return false;
+}
+// ! [function_pass:serialize_cpp]
```

Comment:
I think we don't need this doxygen tags

Reply:
Removed in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/1be8c85bc06795554e80cf5b9903214b2f2248e7)